pytorch_lightning/profiler/pytorch.py

```diff
@@ -142,8 +145,9 @@ def _step(self) -> None:
         if self._current_action == "training_step_and_backward":
             self._num_training_step_and_backward += 1
         elif self._current_action == "validation_step":
-            if self._start_action_name == "on_train_start" and self._num_training_step_and_backward > 0:
-                self._num_validation_step += 1
+            if self._start_action_name == "on_fit_start":
+                if self._num_training_step_and_backward > 0:
```

Comment:
Unrelated to this PR and just my own curiosity, could you explain why we need to checkout `if self._num_training_step_and_backward > 0:`

Reply:
Because we don't want to record during sanity_checking.

Reply:
Could you add this as a comment in the PR?